nwbwidgets/controllers/time_window_controllers.py

```diff
@@ -1,101 +1,102 @@
+        **kwargs,
     ):
-
-        if orientation not in ("horizontal", "vertical"):
-            ValueError("Unrecognized orientation: {}".format(orientation))
-
-        super().__init__()
         self.vmin = vmin
         self.vmax = vmax
         self.start_value = start_value
         self.description = description
         self.orientation = orientation
 
+        super().__init__()
+
+    def setup_attributes(self):
         if self.orientation == "horizontal":
             self.to_start_button = widgets.Button(description="◀◀", layout=Layout(width="65px"))
             self.backwards_button = widgets.Button(description="◀", layout=Layout(width="40px"))
             self.forward_button = widgets.Button(description="▶", layout=Layout(width="40px"))
             self.to_end_button = widgets.Button(description="▶▶", layout=Layout(width="65px"))
-        else:  # vertical
+        elif self.orientation == "vertical":
             self.to_end_button = widgets.Button(description="▲▲", layout=Layout(width="50px"))
             self.forward_button = widgets.Button(description="▲", layout=Layout(width="50px"))
             self.backwards_button = widgets.Button(description="▼", layout=Layout(width="50px"))
             self.to_start_button = widgets.Button(description="▼▼", layout=Layout(width="50px"))
+        else:
+            raise ValueError(
+                f"Unrecognized orientation '{self.orientation}', should be either 'horizontal' or 'vertical'"
+            )
 
-        self.to_start_button.on_click(self.move_start)
-        self.backwards_button.on_click(self.move_down)
-        self.forward_button.on_click(self.move_up)
-        self.to_end_button.on_click(self.move_end)
-
-    # @abstractmethod
-    # @property
-    # def value(self):
-    #    """Must be window in seconds"""
-    #    pass
-
-    @abstractmethod
-    def get_children(self):
-        pass
-
-    @abstractmethod
     def move_up(self, change):
-        pass
+        raise NotImplementedError("The 'move_up' method of this WindowController has not been defined.")
 
-    @abstractmethod
     def move_down(self, change):
-        pass
+        raise NotImplementedError("The 'move_down' method of this WindowController has not been defined.")
 
-    @abstractmethod
     def move_start(self, change):
-        pass
+        raise NotImplementedError("The 'move_start' method of this WindowController has not been defined.")
 
-    @abstractmethod
     def move_end(self, change):
-        pass
+        raise NotImplementedError("The 'move_end' method of this WindowController has not been defined.")
+
+    def setup_observers(self):
+        self.to_start_button.on_click(self.move_start)
+        self.backwards_button.on_click(self.move_down)
+        self.forward_button.on_click(self.move_up)
+        self.to_end_button.on_click(self.move_end)
+
+    def get_children(self) -> widgets.Box:
```

Comment:
I'm not entirely sure if this method is ever used in a static way, though as written it certainly could.

I had originally thought of controllers and their methods as modifying things entirely in-place